packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/Button.java

```diff
@@ -96,6 +96,8 @@ public static final class Builder {
 
         private List<Diagnostic> diagnostics;
 
+        private Supplier<String> helpTextProvider;
```

Comment:
> You could have simply used a boolean to carry the information relevant for the rendering.

Yes, but I would still have needed about the same amount of changes to carry the boolean (instead of the `Supplier<String>`), wouldn't I?

> I don't see any issues with having some simple inheritance for builders like that since builders are 99% data structure.

AFAICT, it's forbidden by the rules you setup: `immutableClassesShouldHaveANestedBuilder`, and `HaveAValidBuilderCondition` requires the builder class to be `final`.

Reply:
> When the frontend would ask for the help text, you would just ask this service which could then leverage the view model

What about widgets created programatically which do not come from a View model?